sshuttle/methods/nft.py

```diff
@@ -21,16 +22,19 @@ def setup_firewall(self, port, dnsport, nslist, family, subnets, udp,
         def _nft(action, *args):
             return nft(family, table, action, *args)
 
-        chain = 'sshuttle-%s' % port
-
         # basic cleanup/setup of chains
         _nft('add table', '')
-        _nft('add chain', 'prerouting',
-             '{ type nat hook prerouting priority -100; policy accept; }')
-        _nft('add chain', 'postrouting',
-             '{ type nat hook postrouting priority 100; policy accept; }')
-        _nft('add chain', 'output',
-             '{ type nat hook output priority -100; policy accept; }')
+        # prerouting, postrouting, and output chains may already exist
+        for chain in ['prerouting', 'postrouting', 'output']:
+            rules = '{{ type nat hook {} priority -100; policy accept; }}' \
+                    .format(chain)
+            try:
+                _nft('add chain', chain, rules)
+            except Fatal:
```

Comment:
This makes me a little bit nervous, any error will result in this action, even if the error occurred due to unrelated reasons. The nat method uses a `ipt_chain_exists` function from `linux.py` - would it be possible to do something similar here?

Reply:
Fixed and pushed up.